### PR TITLE
Pundit install and configured

### PR DESCRIPTION
**Pull request by FVolonnino**
https://trello.com/c/fb6UGV6A/12-instalar-y-configurar-pundit

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -59,6 +59,13 @@ gem 'sidekiq', '~> 5.2.2'
 
 gem 'redis-rails', '~> 5.0.2'
 
+gem 'dotenv-rails', '~> 2.5.0', groups: [:development, :test]
```
[the remark this answers is not in the document]
Comment (FVolonnino) [reply Gemfile]: The rebases keep including it and I keep forgetting to take it out. I'll try to fix it.

File: Gemfile [hunk 2]
```diff
@@ -59,6 +59,13 @@ gem 'sidekiq', '~> 5.2.2'
...
+
+gem 'activeadmin', '~> 1.3.1'
```
[the remark this answers is not in the document]
Comment (FVolonnino) [reply Gemfile]: Why shouldn't active admin be in this PR?

File: app/controllers/api/v1/rent_controller.rb [hunk 3]
```diff
@@ -20,6 +20,12 @@ def create
         end
       end
 
+      def show
```
[the remark this answers is not in the document]
Comment (FVolonnino) [reply app/controllers/api/v1/rent_controller.rb]: No, but I needed an unsafe endpoint to test pundit. Index already only shows the rents belonging to the current user so there's nothing I can authorize there. I could make an authorization so users can only create rents belonging to them but I didn't think about that yesterday.

File: app/policies/rent_policy.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class RentPolicy
```
[the remark this answers is not in the document]
Comment (FVolonnino) [reply app/policies/rent_policy.rb]: I'll add the one for create now. In the comment above this one I explained why I didn't believe Index required authorization.

File: app/controllers/api/v1/rent_controller.rb [hunk 5]
```diff
@@ -3,15 +3,15 @@ module V1
     class RentController < ApiController
       layout 'mailer'
       def index
-        if current_user
-          render_paginated Rent.where(user_id: current_user.id)
-        else
-          render json: 'Error: cannot find user'
+        Rent.where(user_id: current_user.id).each do |rent|
```
Comment (Gmarn) [inline app/controllers/api/v1/rent_controller.rb]: This not the most efficient but works.

File: app/controllers/api/v1/rent_controller.rb [hunk 6]
```diff
@@ -3,15 +3,15 @@ module V1
...
+          authorize rent
         end
+        render_paginated Rent.where(user_id: current_user.id)
       end
 
       def create
         rent = Rent.new(create_params)
+        authorize rent
```
Comment (Gmarn) [inline app/controllers/api/v1/rent_controller.rb]: Ok, now we have a way to authorize endpoints. How did you manage the `:unauthorized` error? Maybe should be good using a `rescue_from` in your `ApiController` to handle the error?

File: app/controllers/api/v1/api_controller.rb [hunk 7]
```diff
@@ -6,6 +6,12 @@ class ApiController < ApplicationController
       protect_from_forgery with: :null_session
       before_action :authenticate_user!
       before_action :set_locale
+      rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+      def user_not_authorized
+        flash[:warning] = "You have met a terrible fate, haven't you?"
+        redirect_to '/api/v1/book'
```
Comment (Gmarn) [inline app/controllers/api/v1/api_controller.rb]: Stange URL to redirect, but ok!

Comment (Gmarn) [reply app/controllers/api/v1/api_controller.rb]: There is other syntaxis from rails to url and path

Comment (FVolonnino) [reply app/controllers/api/v1/api_controller.rb]: I don't have a root so I had to think of something else to redirect to-

**Pull request merged**